lib/api/objectCopy.js

```diff
@@ -217,10 +219,21 @@ function objectCopy(authInfo, request, sourceBucket,
                 return next(null, storeMetadataParams, dataLocator, destObjMD,
                     serverSideEncryption, destBucketMD);
             }
-             // dataLocator is an array.  need to get and put all parts
-             // For now, copy 1 part at a time. Could increase the second
-             // argument here to increase the number of parts
-             // copied at once.
+
+            const backendInfoObj = locationConstraintCheck(request,
+                storeMetadataParams.metaHeaders, destBucketMD, log);
+            if (backendInfoObj.err) {
+                return process.nextTick(() => {
```

Comment:
I would get rid of the process.nextTick() here since we're already inside async.waterfall which is controlling the error callback behavior behind the scenes.